### PR TITLE
Remove a few circumvolutions around parameters of inductive entries

### PR DESCRIPTION
**Pull request by maximedenes**
This PR is trying to clean a bit the kernel interfaces, by making input and output data closer (output data should be input data that has been checked + derived data).

It is very much work in progress, there's only a tiny clean-up for now, but feedback on the general idea is welcome.

### CODE REVIEW

File: kernel/typeops.ml [hunk 1]
```diff
@@ -482,25 +482,19 @@ let infer_v env cv =
...
+    match d with
+      | LocalAssum (_,ty) ->
+        let _ = infer_type env ty in
+        push_rel d env
+      | LocalDef (_,bd,ty) ->
+        let j1 = infer env bd in
+        let _ = infer_type env ty in
+        conv_leq false env j1.uj_type ty;
+        push_rel d env)
+    rels ~init:env
```
Comment (SkySkimmer) [inline kernel/typeops.ml]: I believe this drops the check_wellformed_universes.

**Review by SkySkimmer — APPROVED** [review]
So this is removing LocalFooEntry?
+1

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
Actually let me request changes over my previous comment, it shouldn't just be dropped ;)

File: kernel/typeops.ml (hunk 1 above)
Comment (maximedenes) [reply kernel/typeops.ml]: Isn't done as part of `infer_type`?

Comment (SkySkimmer) [reply kernel/typeops.ml]: Oh you're using infer_type
OK then

**Review by SkySkimmer — APPROVED** [review]

**Review by mattam82 — APPROVED** [review]
Nice! I'm wondering why we ever wanted a separate type for the params.

**Review by herbelin — APPROVED** [review]
Had a quick look, seems good and relevant.

**Comment by SkySkimmer** [comment]
By the way this means that 
~~~coq
Definition nat' := nat.
Inductive foo (A:=0:nat') := .
Print foo.
~~~
should now print `nat'` instead of `nat`.

**Comment by herbelin** [comment]
@SkySkimmer: would you assign?

**Comment by SkySkimmer** [comment]
>@SkySkimmer: would you assign?

I'm fine with doing it but why me?

**Comment by herbelin** [comment]
> I'm fine with doing it but why me?

No strong reasons. First because you commented and made good remarks about this PR. Second because @mattam82 tends to be overbusy (but maybe that's changing). Third, for rotating the responsibilities, but maybe you think that you are already doing a lot.

**Comment by SkySkimmer** [comment]
>maybe you think that you are already doing a lot.

I said I'm fine though.


In any case we have to wait for overlays.

**Comment by gares** [comment]
I'm doing the one for elpi

**Comment by gares** [comment]
There it is
  https://github.com/LPCIC/coq-elpi/tree/overlay/8684
but I think this PR needs a rebase since I had to "undo" the last PR being merged (monic hooks by Emilio)

**Pull request merged**